pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -202,3 +203,200 @@ def adjoint(self):
         op = TClock(wires=self.wires)
         op.inverse = not self.inverse
         return op
+
+
+class TAdd(Operation):
+    r"""TAdd(wires)
+    The 2-qutrit controlled add gate
+
+    The construction of this operator is based on definition 7 from
+    `Yeh et al. (2022) <https://arxiv.org/abs/2204.00552>`_.
+
+    .. math:: TAdd = \begin{bmatrix}
+                        1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\
+                        0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\
+                        0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 \\
+                        0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 \\
+                        0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 \\
+                        0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 \\
+                        0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 \\
+                        0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 \\
+                        0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0
+                    \end{bmatrix}
+
+    .. note:: The first wire provided corresponds to the **control qutrit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+    """
+    num_wires = 2
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    @staticmethod
+    def compute_matrix():
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.TAdd.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.TAdd.compute_matrix())
```

Comment:
One thing that came to my mind afterwards:
Thinking for simplicity in terms of CNOT, the matrix representation can look different depending on whether the first or the second qubit acts as a control. Is this in some way being taken into account?
![image](https://user-images.githubusercontent.com/43949391/186205024-c24c2ca6-6530-4a6f-abd2-0342cb446cc1.png)

Reply:
Indeed, this is related to the above comment about textbook matrix representation and wire order. `Operation.compute_matrix` gives only the canonical representation, but one can call `Operation.matrix` which takes a `wire_order` parameter (which works by first calling `compute_matrix` and then `expand_matrix` if necessary to adjust the ordering).

Incidentally, `expand_matrix` actually has to be adjust to work for qutrit operations, which is going to be taken care of in a separate PR.